File: Pedometer/app/src/main/java/com/example/pedometer/MainActivity.java
package com.example.pedometer;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void navigateToStepCounterActivity(View v) {
        Intent mIntent = new Intent(this, StepsCounterActivity.class);
        startActivity(mIntent);
    }

    public void navigateToStepHistoryActivity(View v) {
        Intent mIntent = new Intent(this, StepsHistoryActivity.class);
        startActivity(mIntent);
    }

    public void navigateToCustomAlgoActivity(View v) {
        Intent mIntent = new Intent(this, CustomAlgoResultsActivity.class);
        startActivity(mIntent);

    }
}

File: DataBindingObject/app/src/main/java/com/example/databindingobject/MainActivity.kt
package com.example.databindingobject

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.example.databindingobject.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.student = getStudent()
    }

    private fun getStudent():Student {
        return Student(1, "Edward", "ed@gmail.com")
    }

}
